misc/tools.php

```diff
@@ -1,9 +1,16 @@
 <?php
     function get_base_url($url) {
         $parsed = parse_url($url);
-        return $parsed["scheme"] . "://" . $parsed["host"] . "/";
+
+        if (isset($parsed["scheme"]) && isset($parsed["host"]) && !empty($parsed["scheme"]) && !empty($parsed["host"])) {
```

Comment:
Hmm.. On second thought,

`if (!empty($parsed["scheme"]) && !empty($parsed["host"])) {` would be enough

>isset() would return true if a variable is set and not NULL, regardless of its content.
empty() would return true if a variable is set and is considered empty (e.g., an empty string "", 0, NULL, false, or an empty array), otherwise it returns false.